Traceback (most recent call last):
  File "/Users/johnmyers/opt/anaconda3/envs/python-2023-08-v2/lib/python3.11/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/Users/johnmyers/opt/anaconda3/envs/python-2023-08-v2/lib/python3.11/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/johnmyers/opt/anaconda3/envs/python-2023-08-v2/lib/python3.11/site-packages/jupyter_core/utils/__init__.py", line 166, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/johnmyers/opt/anaconda3/envs/python-2023-08-v2/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Users/johnmyers/opt/anaconda3/envs/python-2023-08-v2/lib/python3.11/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/Users/johnmyers/opt/anaconda3/envs/python-2023-08-v2/lib/python3.11/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/Users/johnmyers/opt/anaconda3/envs/python-2023-08-v2/lib/python3.11/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# instantiate a JointGrid object
g = sns.JointGrid()

# generate the plots
scatter = sns.scatterplot(data=joint_data, x='X', y='Y', ax=g.ax_joint, alpha=0.05,)
sns.scatterplot(data=slice_data, x='X', y='Y', ax=g.ax_joint)
sns.histplot(data=slice_data, x='X', ax=g.ax_marg_x, ec='b', color='#FD46FC', kde=True)

g.ax_marg_y.remove()
g.set_axis_labels(xlabel=r'$x=$size (ft$^2$)', ylabel=r'$y=$price (\$1,000)')
plt.tight_layout()
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mNameError[0m                                 Traceback (most recent call last)
Cell [0;32mIn[8], line 5[0m
[1;32m      2[0m g [38;5;241m=[39m sns[38;5;241m.[39mJointGrid()
[1;32m      4[0m [38;5;66;03m# generate the plots[39;00m
[0;32m----> 5[0m scatter [38;5;241m=[39m sns[38;5;241m.[39mscatterplot(data[38;5;241m=[39m[43mjoint_data[49m, x[38;5;241m=[39m[38;5;124m'[39m[38;5;124mX[39m[38;5;124m'[39m, y[38;5;241m=[39m[38;5;124m'[39m[38;5;124mY[39m[38;5;124m'[39m, ax[38;5;241m=[39mg[38;5;241m.[39max_joint, alpha[38;5;241m=[39m[38;5;241m0.05[39m,)
[1;32m      6[0m sns[38;5;241m.[39mscatterplot(data[38;5;241m=[39mslice_data, x[38;5;241m=[39m[38;5;124m'[39m[38;5;124mX[39m[38;5;124m'[39m, y[38;5;241m=[39m[38;5;124m'[39m[38;5;124mY[39m[38;5;124m'[39m, ax[38;5;241m=[39mg[38;5;241m.[39max_joint)
[1;32m      7[0m sns[38;5;241m.[39mhistplot(data[38;5;241m=[39mslice_data, x[38;5;241m=[39m[38;5;124m'[39m[38;5;124mX[39m[38;5;124m'[39m, ax[38;5;241m=[39mg[38;5;241m.[39max_marg_x, ec[38;5;241m=[39m[38;5;124m'[39m[38;5;124mb[39m[38;5;124m'[39m, color[38;5;241m=[39m[38;5;124m'[39m[38;5;124m#FD46FC[39m[38;5;124m'[39m, kde[38;5;241m=[39m[38;5;28;01mTrue[39;00m)

[0;31mNameError[0m: name 'joint_data' is not defined

